deployment/deployment-config.ts

```diff
@@ -0,0 +1,96 @@
+import { isLocalStack } from "./local/util";
+import { getSecret } from "./utils/secrets-manager";
+
+export interface DeploymentConfigProperties {
+  project: string;
+  stage: string;
+  isDev: boolean;
+  vpcName: string;
+  oktaMetadataUrl: string;
+  bootstrapUsersPassword?: string;
+  launchDarklyClient: string;
+  redirectSignout: string;
+  cloudfrontCertificateArn?: string;
+  cloudfrontDomainName?: string;
+  secureCloudfrontDomainName?: string;
+  userPoolDomainPrefix?: string;
+  vpnIpSetArn?: string;
+  vpnIpv6SetArn?: string;
+  brokerString: string;
+  kafkaAuthorizedSubnetIds: string;
+}
+
+export const determineDeploymentConfig = async (stage: string) => {
+  const project = process.env.PROJECT!;
+  const isDev =
+    isLocalStack ||
+    !["master", "main", "val", "production", "jon-cdk"].includes(stage); // TODO: remove jon-cdk after main is deployed
+  const secretConfigOptions = {
+    ...(await loadDefaultSecret(project, stage)),
+    ...(await loadStageSecret(project, stage)),
+  };
+
+  const config = {
+    project,
+    stage,
+    isDev,
+    ...secretConfigOptions,
+  };
+  if (config.cloudfrontDomainName) {
+    config.secureCloudfrontDomainName = `https://${config.cloudfrontDomainName}/`;
+  }
+
+  if (!isLocalStack) {
+    validateConfig(config);
+  }
+
+  return config;
+};
+
+export const loadDefaultSecret = async (
+  project: string,
+  stage: string = "bootstrap"
+) => {
+  if (["bootstrap", "prerequisites"].includes(stage)) {
```

Comment:
I only had to do this for bootstrap for some reason. Maybe that's just because I'd already put the secret in dev before I ran the prerequisites. Either is fine but I'm tempted to standardize one way or the other.

Reply:
I agree, let's standardize one way or the other.